### PR TITLE
Fix getAssetInfoAsync does not return localUri for videos

### LINKED ISSUE #2448: [ios] MediaLibrary.getAssetInfoAsync does not return localUri for mediaType "video"
Environment:
  OS: macOS 10.14
  Node: 10.10.0
  Yarn: Not Found
  npm: 6.4.1
  Watchman: 4.9.0
  Xcode: Xcode 10.0 Build version 10A255
  Android Studio: 3.2 AI-181.5540.7.32.5056338

Packages: (wanted => installed)
  expo: ^30.0.1 => 30.0.1
  react: 16.3.1 => 16.3.1
  react-native: https://github.com/expo/react-native/archive/sdk-30.0.0.tar.gz => 0.55.4

Calling MediaLibrary.getAssetInfoAsync for a media library asset of type "video" does not return a localUri. The documentation does not give any indication that this method should return a different object shape to type 'photo', so I assume it is a bug.

I can't get Snack to play nicely with this for whatever reason, but here is code to reproduce the error:

```javascript
'use strict';
import React, { Component } from 'react';
import { Button } from 'react-native';
import { MediaLibrary, Permissions, Camera } from 'expo';

class CameraScreen extends Component {
    state = { recording: false };

    componentDidMount() {
        Permissions.askAsync(
            Permissions.CAMERA,
            Permissions.CAMERA_ROLL,
            Permissions.AUDIO_RECORDING
        );
    }

    takeVideo = async () => {
        if (this.camera && !this.state.recording) {
            this.setState({ recording: true });
            this.video = this.camera.recordAsync();
        } else if (this.camera && this.state.recording) {
            this.setState({ recording: false });
            this.camera.stopRecording();
            const { uri } = await this.video;
            const asset = await MediaLibrary.createAssetAsync(uri);
            const assetInfo = await MediaLibrary.getAssetInfoAsync(asset);
            console.log(assetInfo);
        }
    };

    render() {
        return (
            <Camera
                ref={ref => (this.camera = ref)}
                style={{ flex: 1, justifyContent: 'center' }}
            >
                <Button
                    onPress={this.takeVideo}
                    title={
                        this.state.recording ? 'Stop recording' : 'Start recording'
                    }
                />
            </Camera>
        );
    }
}

export default CameraScreen;
```

Here is what the above will log. As you can see, there is no localUri:

```javascript
Object {
  "creationTime": 1539848674000,
  "duration": 2.868333333333333,
  "filename": "IMG_1566.MOV",
  "height": 1920,
  "id": "B4AEEE95-32F2-4968-94B7-BFB1B68E694C/L0/001",
  "isFavorite": false,
  "isHidden": false,
  "location": null,
  "mediaSubtypes": Array [],
  "mediaType": "video",
  "modificationTime": 1539848677094,
  "orientation": 0,
  "uri": "assets-library://asset/asset.MOV?id=B4AEEE95-32F2-4968-94B7-BFB1B68E694C&ext=MOV",
  "width": 1080,
}
```

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Resolve #2448.

# How

For video files, to get additional information we need to use `requestAVAssetForVideo` method.

# Test Plan

- https://snack.expo.io/@lukaszkosmaty/medialibrary.getassetinfoasync-does-not-return-localuri-for-mediatype-%22video%22 - now display `local Uri` property for both images and videos.

### CODE REVIEW

File: packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m [hunk 1]
```diff
@@ -388,19 +388,31 @@ - (NSDictionary *)constantsToExport
...
+      }];
+    } else {
+      PHVideoRequestOptions *options = [PHVideoRequestOptions new];
+      options.networkAccessAllowed = YES;
+
+      [[PHImageManager defaultManager] requestAVAssetForVideo:asset
+                                                      options:options
+                                                resultHandler:^(AVAsset * _Nullable asset, AVAudioMix * _Nullable audioMix, NSDictionary * _Nullable info) {
+        AVURLAsset *urlAsset = (AVURLAsset *)asset;
+        result[@"localUri"] = [[urlAsset URL] absoluteString];
```
Comment (tsapeta) [inline packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m]: Is there anything else available in AVURLAsset that we can export to JS?
Also, it might be worth to update the documentation as well that we don't provide `orientation` when the `mediaType` is different than `photo`

Comment (lukmccall) [reply packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m]: `AVUrlAssets` doesn't provide anything useful.

In our documentation, in the description column of `orientation` property, we have this:
`Display orientation of the image`. Is it not enough?

Comment (tsapeta) [reply packages/expo-media-library/ios/EXMediaLibrary/EXMediaLibrary.m]: I'd rephrase it to make it clearer for users: 
> Orientation is available only for assets whose `mediaType` is `MediaType.photo`
Also, maybe we can infer the orientation based on the video dimensions? 🤔
